internal/host/plugin/repository_host_catalog_test.go

```diff
@@ -368,12 +369,736 @@ func TestRepository_CreateCatalog(t *testing.T) {
+			assert.NoError(
+				db.TestVerifyOplog(
+					t,
+					dbRW,
+					gotCatalog.PublicId,
+					db.WithOperation(op),
+					db.WithCreateNotBefore(10*time.Second),
+				),
+			)
+		}
+	}
+
+	tests := []struct {
+		name               string
+		withEmptyPluginMap bool
+		withRespSecretsNil bool
+		withPluginError    error
+		changeFuncs        []changeHostCatalogFunc
+		version            uint32
+		fieldMask          []string
+		wantCheckFuncs     []checkFunc
+		wantIsErr          errors.Code
+	}{
+		{
+			name:        "nil catalog",
+			changeFuncs: []changeHostCatalogFunc{changeCatalogToNil()},
+			wantIsErr:   errors.InvalidParameter,
+		},
+		{
+			name:        "nil embedded catalog",
+			changeFuncs: []changeHostCatalogFunc{changeEmbeddedCatalogToNil()},
+			wantIsErr:   errors.InvalidParameter,
+		},
+		{
+			name:        "missing public id",
+			changeFuncs: []changeHostCatalogFunc{changePublicId("")},
+			wantIsErr:   errors.InvalidParameter,
+		},
+		{
+			name:        "missing scope id",
+			changeFuncs: []changeHostCatalogFunc{changeScopeId("")},
+			wantIsErr:   errors.InvalidParameter,
+		},
+		{
+			name:      "empty field mask",
+			fieldMask: nil, // Should be testing on len
+			wantIsErr: errors.EmptyFieldMask,
+		},
+		{
+			name:        "bad catalog id",
+			changeFuncs: []changeHostCatalogFunc{changePublicId("badid")},
+			fieldMask:   []string{"name"},
+			wantIsErr:   errors.RecordNotFound,
+		},
+		{
+			name:        "version mismatch",
+			changeFuncs: []changeHostCatalogFunc{changeName("foo")},
+			version:     1,
+			fieldMask:   []string{"name"},
+			wantIsErr:   errors.VersionMismatch,
+		},
+		{
+			name:               "plugin lookup error",
+			withEmptyPluginMap: true,
+			version:            2,
+			fieldMask:          []string{"name"},
+			wantIsErr:          errors.Internal,
+		},
+		{
+			name:            "plugin invocation error",
+			withPluginError: errors.New(context.Background(), errors.Internal, "TestRepository_UpdateCatalog/plugin_invocation_error", "test plugin error"),
+			version:         2,
+			fieldMask:       []string{"name"},
+			wantIsErr:       errors.Internal,
+		},
+		{
+			name:        "update name (duplicate, same scope)",
+			changeFuncs: []changeHostCatalogFunc{changeName(testDuplicateCatalogNameProjectScope)},
+			version:     2,
+			fieldMask:   []string{"name"},
+			wantIsErr:   errors.NotUnique,
+		},
+		{
+			name:        "update name",
+			changeFuncs: []changeHostCatalogFunc{changeName("foo")},
+			version:     2,
+			fieldMask:   []string{"name"},
+			wantCheckFuncs: []checkFunc{
+				checkVersion(3),
+				checkUpdateCatalogRequestCurrentNameNil(),
+				checkUpdateCatalogRequestNewName("foo"),
+				checkName("foo"),
+				checkSecrets(map[string]interface{}{
+					"one": "two",
+				}),
+				checkNumUpdated(1),
+				checkVerifyCatalogOplog(oplog.OpType_OP_TYPE_UPDATE),
+			},
+		},
+		{
+			name:        "update name to same",
+			changeFuncs: []changeHostCatalogFunc{changeName("")},
```

Comment:
This actually fails a db check if it is actually going through.  Names must have length > 0.  see wt_name constraints in internal/db/schema/migrations/postgres/1/02_domains.up.sql.   Is the row actually updated (the update_time being updated?)

Reply:
It _should_ be nulling out as per https://github.com/hashicorp/boundary/pull/1634/files#diff-9e3211879fb53616a797ca22d9549ac9f558ecc6e5ad6ca5b1a16bfcfc5be6b9R222-R224. Should be okay?

Reply:
Discussed with @talanknight and confirmed this was okay.